enhancements/sig-policy/62-namespace-labelselector/README.md

```diff
@@ -0,0 +1,207 @@
+`exclude` arrays. Either the `include` and `exclude` arrays or the `LabelSelector` can be provided.
+If the `LabelSelector` is not provided, all namespaces are retrieved and filtered with the `include`
+and `exclude` arrays. (See the
+[LabelSelector](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#LabelSelector) struct for
+reference.)
+
+### User Stories
+
+Suppose clusters have namespaces named `test1` through `test5` labeled with a `name` key and value
+matching its name and an `all: namespaces` label.
+
+#### Story 1
+
+As an administrator of `ConfigurationPolicies`, I can choose namespaces in which to deploy defined
+objects by the namespace label.
+
+```yaml
+namespaceSelector:
+  matchLabels:
+    name: test2
+  matchExpressions:
+    key: name
+    operator: In
+    values: ["test1", "test2"]
+```
+
+**Result**: `test2`
+
+#### Story 2
+
+As an administrator of `ConfigurationPolicies`, I can use `include`/`exclude` filepath patterns to
+filter out by namespace name, and without needing to provide `MatchLabels`/`MatchExpressions`.
+
+```yaml
+namespaceSelector:
+  include: ["test[3-5]"]
+  exclude: ["test4"]
+```
+
+**Result**: `test3`, `test5`
+
+#### Story 3
+
+As an administrator of `ConfigurationPolicies`, I can choose namespaces in which to deploy defined
+objects by the namespace label and then filter out by name using filepath patterns.
+
+```yaml
+namespaceSelector:
+  exclude: ["test[2-3]"]
+  matchLabels:
+    all: namespaces
+```
+
+**Result**: `test1`, `test4`, `test5`
+
+#### Story 4
+
+As an administrator of `ConfigurationPolicies`, I can override the `NamespaceSelector` by providing
+`metadata.namespace` in the object itself. (This should match current behavior--putting here for
+clarity.)
+
+```yaml
+namespaceSelector:
+  include: ["test?"]
+  matchExpressions:
+    key: all
+    operator: Exists
+object-templates:
+  - objectDefinition:
+    metadata:
+      ...
+      namespace: this-other-namespace
+```
+
+**Result**:
+
+- Namespaces from `namespaceSelector`: `test1`, `test2`, `test3`, `test4`, `test5`
+- Object defined to check namespace: `this-other-namespace`
+
+### Implementation Details/Notes/Constraints
+
+Current behaviors of `NamespaceSelector` won't be disrupted. When namespaces are fetched, if a
+`LabelSelector` is provided, it will be added to the API call to return matching namespaces.
+Filtering by `include`/`exclude` will take place against the fetched namespaces as before. The
+`NamespaceSelector` can be overridden by providing `metadata.namespace` in the object.
+
+### Risks and Mitigation
+
+(unknown)
+
+## Design Details
+
+Implement `MatchLabels` and `MatchExpressions` alongside the `include` and `exclude` arrays:
+
+```yaml
+namespaceSelector:
+  include: []
+  exclude: []
+  matchLabels: {}
+  matchExpressions: []
+```
+
+These can easily be used to construct a `LabelSelector` for the Kubernetes API to consume.
```

Comment:
In this section, instead of repeating the YAML format, can we show what the struct will look like in Go, since that is more detailed. Possibly also how the `LabelSelector` will be created from that struct.

The final working implementation doesn't need to be here, I just think it could help prevent surprises.

Reply:
I'll add it to the implementation details since the `struct` is a bit more technical in nature.

Reply:
Oh, actually, I didn't even realize that wasn't even the section we're in! Well shoot. Maybe I'll take this section out? Either that or move content from Implementation Details...

Reply:
Okay, it's gone. 😄